tests/Models/Multiplayer/ScoreLinkTest.php

```diff
@@ -0,0 +1,165 @@
+<?php
+
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+declare(strict_types=1);
+
+namespace Tests\Models\Multiplayer;
+
+use App\Exceptions\InvariantException;
+use App\Models\Beatmap;
+use App\Models\Multiplayer\PlaylistItem;
+use App\Models\Multiplayer\ScoreLink;
+use App\Models\User;
+use Carbon\Carbon;
+use Tests\TestCase;
+
+class ScoreLinkTest extends TestCase
+{
+    public function testRequiredModsMissing()
+    {
+        $user = User::factory()->create();
+        $beatmap = Beatmap::factory()->create();
```

Comment:
beatmap is already created by the playlist item factory

(and the `complete` function call below can use `$playlistItem->beatmap_id` and `$playlistItem->ruleset_id`)